packages/@azure/servicebus/data-plane/test/renewLockSessions.spec.ts

```diff
@@ -0,0 +1,715 @@
+async function testBatchReceiverManualLockRenewalHappyCase(
+  senderClient: QueueClient | TopicClient,
+  receiverClient: QueueClient | SubscriptionClient
+): Promise<void> {
+  await senderClient.getSender().send(testMessage);
+
+  const sessionClient = await receiverClient.getSessionReceiver({ sessionId: testSessionId });
+  const msgs = await sessionClient.receiveBatch(1);
+
+  // Compute expected initial lock duration
+  const expectedLockExpiryTimeUtc = new Date();
+  expectedLockExpiryTimeUtc.setSeconds(
+    expectedLockExpiryTimeUtc.getSeconds() + lockDurationInMilliseconds / 1000
+  );
+
+  should.equal(Array.isArray(msgs), true);
+  should.equal(msgs.length, 1);
+  should.equal(msgs[0].body, testMessage.body);
+  should.equal(msgs[0].messageId, testMessage.messageId);
+
+  // Verify actual lock duration is reset
+  assertTimestampsAreApproximatelyEqual(
+    sessionClient.sessionLockedUntilUtc,
+    expectedLockExpiryTimeUtc,
+    "Initial"
+  );
+
+  // Sleeping 10 seconds...
+  await delay(10000);
+
+  await sessionClient.renewLock();
+
+  // Compute expected lock duration after 10 seconds of sleep
+  expectedLockExpiryTimeUtc.setSeconds(expectedLockExpiryTimeUtc.getSeconds() + 10);
+
+  // Verify actual lock duration is reset
+  assertTimestampsAreApproximatelyEqual(
+    sessionClient.sessionLockedUntilUtc,
+    expectedLockExpiryTimeUtc,
+    "After first renewal"
+  );
+
+  // Sleeping 5 more seconds...
+  await delay(5000);
+
+  await sessionClient.renewLock();
+
+  // Compute expected lock duration after 5 more seconds of sleep
+  expectedLockExpiryTimeUtc.setSeconds(expectedLockExpiryTimeUtc.getSeconds() + 5);
+
+  // Verify actual lock duration is reset
+  assertTimestampsAreApproximatelyEqual(
+    sessionClient.sessionLockedUntilUtc,
+    expectedLockExpiryTimeUtc,
+    "After second renewal"
+  );
+
+  await msgs[0].complete();
+}
+
+// Receive a msg using Batch Receiver, wait until its lock expires, completing it now results in error
+async function testBatchReceiverManualLockRenewalErrorOnLockExpiry(
+  senderClient: QueueClient | TopicClient,
+  receiverClient: QueueClient | SubscriptionClient
+): Promise<void> {
+  await senderClient.getSender().send(testMessage);
+
+  let sessionClient = await receiverClient.getSessionReceiver({ sessionId: testSessionId });
+  const msgs = await sessionClient.receiveBatch(1);
+
+  should.equal(Array.isArray(msgs), true);
+  should.equal(msgs.length, 1, "Expected message length does not match");
+  should.equal(msgs[0].body, testMessage.body);
+  should.equal(msgs[0].messageId, testMessage.messageId);
+
+  // Sleeping (maxAutoRenewDurationInSeconds + lockDuration) seconds...
+  await delay(maxAutoRenewDurationInSeconds * 1000 + lockDurationInMilliseconds + 1000);
+
+  let errorWasThrown: boolean = false;
+  await msgs[0].complete().catch((err) => {
+    should.equal(err.name, "Error");
+    should.equal(!(err.message.search("Cannot find the receiver with name") + 1), false);
```

Comment:
Would be nice if we can have SDK throw SessionLockLostError, but maybe that's too much for this PR

Reply:
Interesting... So we get the `MessageLockLostError` for losing lock on simple messages. But that seems to come from the service and not our SDK, correct?